xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -148,7 +148,15 @@ public String toString() {
       this.subZone = subZone;
     }
 
-    static Locality fromEnvoyProtoLocality(io.envoyproxy.envoy.api.v2.core.Locality locality) {
+    static Locality fromEnvoyProtoLocality(io.envoyproxy.envoy.config.core.v3.Locality locality) {
+      return new Locality(
+          /* region = */ locality.getRegion(),
+          /* zone = */ locality.getZone(),
+          /* subZone = */ locality.getSubZone());
+    }
+
+    @VisibleForTesting
+    static Locality fromEnvoyProtoLocalityV2(io.envoyproxy.envoy.api.v2.core.Locality locality) {
```

Comment:
We should either use all v2 or all v3 messages for tests, but not a mix of both (although both can be handled). But we should just choose one version for all unit tests.

Reply:
V2 messages are used in the tests with a server sending response. We only test server sending v2 messages. There is no mix. Some v3 messages in the tests are testing EnvoyProtoData and some utils methods, not server response.